vignettes/faq.Rmd

```diff
@@ -0,0 +1,183 @@
+  dplyr[collect, tbl]
+  shiny[moduleServer, NS],
+  pool[dbPool],
+  RSQLite[SQLite],
+)
+
+# connection would usually go to `global.R`
+connection <- dbPool(
+  SQLite(),
+  dbname = "database.sqlite"
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+
+  ...
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    ...
+
+    connection |>
+      tbl("data") |>
+      collect()
+  })
+}
+```
+
+</details>
+
+<details>
+<summary>Where is `www` directory?</summary>
+
+Instead of `www`, Rhino uses `app/static`.
+To utilize it, you need to provide the path to your assets that include `static`,
+for example:
+
+```
+img(src = "static/images/appsilon-logo.png")
+```
+
+</details>
+
+# Common problems
+
+<details>
+<summary>Why my `input` does not trigger the reactive chain?</summary>
+
+Rhino uses Shiny modules to encapsulate the whole application,
+so the `id` of every input you create needs to be wrapped in `ns`, for example:
+
+```r
+box::use(
+  shiny[NS, textInput],
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+
+  textInput(
+    inputId = ns("my_input"),
+    label = "My Input"
+  )
+}
+```
+
+</details>
+
+<details>
+<summary>Why my `output` is not visible?</summary>
+
+Rhino uses Shiny modules to encapsulate the whole application,
+so the `id` of every output you create needs to be wrapped in `ns`, for example:
+
+```r
+box::use(
+  shiny[moduleServer, NS, renderText, textOutput],
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+
+  textOutput(ns("text"))
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    output$text <- renderText("This is a message!")
+  })
+}
+
+```
+
+</details>
+
+<details>
+<summary>`shinyBS` is not working in my Rhino application.</summary>
+
+`shinyBS` uses the `.onAttach()` hook to call `shiny::addResourcePath()` which is necessary
+for its resources (JS and CSS) to be loaded correctly.
+This hook is never run with `box::use()` (normally a `library()` call would run it).
+It is worth noting that the problem will be also present in vanilla Shiny
+if you only use `::` to access `shinyBS` functions (`shinyBS` should use `.onLoad()` instead).
+
+_Workaround:_ Add the following snippet to your `app/main.R` (just below the `box::use()` statements is a good place):
+
+```r
+# Run the `.onAttach` hook (shinyBS uses it to add a Shiny resource path).
+suppressWarnings(library(shinyBS))
+```
+
+You'll still need to explicitly `box::use()` whatever functions you need from `shinyBS` (`library()` doesn't work in a box module).
+
+More details can be found [here](https://github.com/Appsilon/rhino/discussions/279).
+
+</details>
+
+# Styling Rhino application
+
+<details>
+<summary>Can I use multiple Sass files?</summary>
+
+Yes, you can have any file/directory structure in `app/styles` that you desire.
+The `app/styles/main.scss` is the entrypoint: you'll need to `@use` the other files from `main.scss`.
```

Comment:
Link to [`@use` documentation](https://sass-lang.com/documentation/at-rules/use)?